internal/report/store_test.go

```diff
@@ -222,9 +219,7 @@ func TestSaveReports(t *testing.T) {
 
 	t.Run("Save PolicyReport (create)", func(t *testing.T) {
 		//nolint
-		// fake client has been undeprecated due to overwhemling feedback
-		// https://github.com/kubernetes-sigs/controller-runtime/pull/1101
-		cl := fake.NewFakeClientWithScheme(customScheme, &npr.PolicyReport, &cpr.ClusterPolicyReport)
+		cl := fake.NewClientBuilder().WithScheme(customScheme).WithObjects(&npr.PolicyReport, &cpr.ClusterPolicyReport).Build()
```

Comment:
maybe not related, why do we have the `nolint` there? was that because fake client was temporarily marked as deprecated (hence a linter error)?

Can we remove the `// nolint` directive or make it more specific (as in, make explicit which linter check is disabled)?

This comment applies to some other parts of this file